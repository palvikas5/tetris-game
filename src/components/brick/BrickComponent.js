import React from "react";
import Brick from "./BrickBlock";

export const BrickComponent = ({ grid }) => {
  return (
    <div className="bricks-container">
      {grid.map(rows => {
        return rows.map(brick => {
          if (!brick) return null;
          const key = `${brick.row}-${brick.col}`;

          return (
            <Brick
              key={key}
              row={brick.row}
              col={brick.col}
              bgColor={brick.bgColor}
            />
          );
        });
      })}
    </div>
  );
};
