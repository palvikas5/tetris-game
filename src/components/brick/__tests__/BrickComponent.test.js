import React from "react";
import { render } from "@testing-library/react";
import { BrickComponent } from "../BrickComponent";

describe("<BrickComponent />", () => {
  it("should render successfully", () => {
    const grid = [[{ row: 0, col: 1, bgColor: "red" }]];
    const { container } = render(<BrickComponent grid={grid} />);
    expect(container).toMatchSnapshot();
  });

  it("should not render empty brick", () => {
    const grid = [[null]];
    const { container } = render(<BrickComponent grid={grid} />);
    expect(container).toMatchSnapshot();
  });
});
