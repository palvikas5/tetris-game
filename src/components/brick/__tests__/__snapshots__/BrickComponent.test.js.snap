// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrickComponent /> should not render empty brick 1`] = `
<div>
  <div
    class="bricks-container"
  />
</div>
`;

exports[`<BrickComponent /> should render successfully 1`] = `
<div>
  <div
    class="bricks-container"
  >
    <div
      class="brick brick-0-1"
      style="background-color: red;"
    />
  </div>
</div>
`;
